### DISCUSSION TITLE
Does OMS support Transformations, as specified in the SSP standard?

### DISCUSSION THREAD

**Discussion by Fredrik1Haider** [OpenModelica/OMSimulator#1215] [Q&A]
In section 4.5.3 in [SSP Standard](https://ssp-standard.org/publications/SSP101/SystemStructureAndParameterization101.pdf) it states that it is possible to declare a transformation in the SSD and/or SSM. For example I may have an input called "ambient_temp", originating from some source, that I would now like to apply an transformation to explore model response. Which I would do in an SSM as follows:

`<ssm:MappingEntry source="ambient_temp" target="model_temp">
<LinearTransformation factor="2" offset="10" /></MappingEntry>`

It is however unclear from the documentation if OSM supports this part of the SSP standard. Is it or is it not supported?

TLDR: Does OMS support the GTransformationChoice Elements?

**Comment by rruusu** [comment]
I am not an authority on this matter, but I encountered the same issue with a connection defined via a linear transformation intended to invert a signal's sign between two connected components. Our specific case involved an FMU representing a mechanical load for a rotational mechanics application; it returned the load as a positive torque signal (in Nm), whereas the connected drivetrain component expected a negative torque to represent a load opposing the shaft's speed.

Upon reviewing the code, it appears that OMSimulator lacks support for the `<LinearTransformation>` element entirely. Its only appearances within the codebase are in the XML schema files.

In my opinion, this is a significant omission from the standard that should be clearly documented. This lack of support is a critical gap, as the only way to circumvent it is to modify the FMU components themselves.

**Comment by arun3688** [comment]
@rruusu @Fredrik1Haider The linear transformation support is not available in OMSimulator yet, But since the SSP concepts are huge we focucssed on the most important needs, as @rruusu says this is an important omission we could implement this support in our new `OMSimulator 3.0`.  @rruusu can you provide a small example on how this `LinearTransformation` should work ?

**Comment by rruusu** [comment]
> can you provide a small example on how this LinearTransformation should work ?

If you mean a sample model, I don't have anything public right now. But it's just an addition of a `LinearTransformation` element to a connection.

For a thorough test, we need all the other types too, and also in parameter mappings. `GTransformationChoice` is described in section 4.5.2 of the SSP 1.0.1 specification.
